tests/unit/test_state.py

```diff
@@ -637,14 +646,61 @@ def test__parse_script_secrets_invalid_secret(exception: Exception, expected_err
     assert expected_error in str(exc)
 
 
-def test__parse_script_secrets_invalid_key_value_pair():
+@pytest.mark.usefixtures("patch_juju_version")
+def test__parse_script_secrets_secret_and_config_set():
+    """
+    arrange: given a config option where both secret and the config is set.
+    act: when _parse_script_secrets is called.
+    assert: InvalidSecretError is raised.
+    """
+    mock_charm = MagicMock()
+    mock_charm.config = {
+        state.SCRIPT_SECRET_ID_CONFIG_NAME: "secret:test-secret-id",
+        state.SCRIPT_SECRET_CONFIG_NAME: "test-secret",
+    }
+
+    with pytest.raises(state.InvalidSecretError) as exc:
+        state._parse_script_secrets(charm=mock_charm)
+
+    assert "Both script-secret and script-secret-id configuration option set." in str(exc)
+
+
+def test__parse_script_secrets_secret_unsupported(monkeypatch: pytest.MonkeyPatch):
```

Comment:
Perhaps parametrise this so we can check the boundary a bit? Like perhaps juju version 3.2 and 2.9?

Reply:
Would 3.2 and 2.9 be worth testing? I'm wondering what worth 2.9 and 3.2 would provide :)

Reply:
The below tests version 2.9, right? I agree that 2.9 is probably not worth testing, I just put it in because you were testing for it and I don't mind too much. I do think version 3.2 should be tested because it is on the boundary

Reply:
Yup, i can update the test case to 3.2 but I don't think testing for ge/le equality checks are required - that should be done on the Juju lib side.